verifier/src/tcb.rs

```diff
@@ -63,31 +152,188 @@ mod tests {
     }
 
     #[test]
-    fn two_signatures_errors() {
-        let raw_info = r#"{"tcb_info":{"id":"SGX","version":3,"fmspc":"00906ED50000"},"signature":"hello","signature":"fail"}"#;
+    fn raw_two_signatures_errors() {
+        let raw_info = r#"{"tcbInfo":{"id":"SGX","version":3,"fmspc":"00906ED50000"},"signature":"hello","signature":"fail"}"#;
         assert!(matches!(
             TcbInfoRaw::try_from(raw_info),
             Err(Error::Serde(_))
         ));
     }
 
     #[test]
-    fn two_tcb_infos_errors() {
-        let raw_info = r#"{"tcb_info":"too many cooks","tcb_info":{"id":"SGX","version":3,"fmspc":"00906ED50000"},"signature":"hello"}"#;
+    fn raw_two_tcb_infos_errors() {
+        let raw_info = r#"{"tcbInfo":"too many cooks","tcbInfo":{"id":"SGX","version":3,"fmspc":"00906ED50000"},"signature":"hello"}"#;
         assert!(matches!(
             TcbInfoRaw::try_from(raw_info),
             Err(Error::Serde(_))
         ));
     }
 
     #[test]
-    fn nested_tcb_info_still_in_parent() {
-        let raw_info = r#"{"tcb_info":{"tcb_info":"nested","version":3,"fmspc":"00906ED50000"},"signature":"hello"}"#;
+    fn raw_nested_tcb_info_still_in_parent() {
+        let raw_info = r#"{"tcbInfo":{"tcb_info":"nested","version":3,"fmspc":"00906ED50000"},"signature":"hello"}"#;
         let tcb_info = TcbInfoRaw::try_from(raw_info).expect("Failed to parse TCB info");
         assert_eq!(tcb_info.signature, "hello");
         assert_eq!(
             tcb_info.tcb_info.get(),
             r#"{"tcb_info":"nested","version":3,"fmspc":"00906ED50000"}"#
         );
     }
+
+    #[test]
+    fn parse_example_tcb_info() {
+        let json = include_str!("../data/tests/example_tcb.json");
+        let tcb_raw = TcbInfoRaw::try_from(json).expect("Failed to parse raw TCB");
+        let tcb_info = TcbInfo::try_from(tcb_raw.tcb_info.get()).expect("Failed to parse TCB info");
+        assert_eq!(tcb_info.id, "SGX");
+        assert_eq!(tcb_info.version, 3);
+        assert_eq!(tcb_info.issue_date, "2022-04-13T09:38:17Z");
+        assert_eq!(tcb_info.next_update, "2022-05-13T09:38:17Z");
+        assert_eq!(tcb_info.fmspc, "50806F000000");
+        assert_eq!(tcb_info.pce_id, "0000");
```

Comment:
For this and the subsequent test I'm on the fence how much value the verification of the members provide, it feels like a wall of text trying to grok the test, but asserting that the try_from() worked felt a bit too loose.